nservicebus/upgrades/7to8/index.md

```diff
@@ -249,4 +246,69 @@ if (extensions.TryGet(out DiscardIfNotReceivedBefore constraint))
+
+### Pipeline context types changes
+
+Throughput the pipeline all contexts have been updated to use `ReadonlyMemory<byte>` (e.g. context for behaviors, stage connectors, and message mutators) and affects the following types:
+
+* MutateIncomingTransportMessageContext
+* MutateOutgoingTransportMessageContext
+* ConnectorContextExtensions
+* IIncomingPhysicalMessageContext
+* IncomingPhysicalMessageContext
+* IOutgoingPhysicalMessageContext
+* OutgoingPhysicalMessageContext
+* SerializeMessageConnector
+
+## Message bodies only available in scope of incoming or outgoing messages
+
+Messages bodies available via `ReadonlyMemory<byte>` are only valid while in scope of incoming or outgoing message processing. Once out of scope the data referenced is no longer valid.
+
+If access is still required outside the processing scope it is required to copy the message body while in scope.
+
+### Message Mutators: Updating of message bodies
+
+The message mutator API for changing the message body has changed. Instead, of `UpdateMessage(byte[] body)` method `MutateIncomingTransportMessageContext` and `MutateOutgoingTransportMessageContext` expose `Body` property of type `ReadonlyMemory<T>`.
+
+Message mutators that replace the whole message body might be better of to be converted to a behavior. Via a [pipeline behavior](/nservicebus/pipeline/manipulate-with-behaviors.md). Via behavior it is possible to reduce allocations via [`ArrayPool<byte>`](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1) or packages like [RecyclableMemoryStream](https://github.com/Microsoft/Microsoft.IO.RecyclableMemoryStream).
+
+### Serializers: Deserialization using ReadOnlyMemory of byte
+
+Deserialization updated to use `ReadOnlyMemory<byte>` instead of `Stream` with  on `Deserialize` API method on `IMessageSerializer`.
+
+```csharp
+object[] Deserialize(ReadOnlyMemory<byte> body, IList<Type> messageTypes = null);
+```
+
+If serializers do not support `ReadOnlyMemory<byte>` usage of `.ToArray` must be avoided as this will copy the data and increases memory allocations. Instead include a shim that provides read-only access to the underlying data or use the `ReadOnlyMemoryExtensions.AsStream` from the [Microsoft.Toolkit.HighPerformance](https://docs.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.extensions.readonlymemoryextensions.asstream).
+
+## Transports
+
+### Transports: Outgoing message TransportOperation API
+
+The outgoing message body passed to the transport via `TransportOperation` has a new constructor:
+
+```csharp
+public TransportOperation(string messageId, DispatchProperties properties, ReadOnlyMemory<byte> body, Dictionary<string, string> headers)
+```
+
+### Transports: Incoming message MessageContext API
+
+A message body passed by the transport to the core using `ReadonlyMemory<byte>` instead of `byte[]`. The `MessageContext` becomes:
```

Comment:
```suggestion
A message body passed by the transport to the core using `ReadOnlyMemory<byte>` instead of `byte[]`. The `MessageContext` becomes:
```